### PR TITLE
feat: Custom z bin search order

### PR DESCRIPTION
**Pull request by LuisFelipeCoelho**
This PR introduces a custom z looping feature, 
which allows us to change the order in which `BinnedSPGroup` search for the Space Points.

In the Main branch the groups are sorted in **phi** and **z** in ascending order. But in Athena **z** follows the order within the vector `{1, 2, 3, 4, 11, 10, 9, 8, 6, 5, 7}`

This PR introduces this same custom looping approach with the vector `zBinsCustomLooping`. If it is not defined, the previous behaviour is preserved.

@noemina @paulgessinger @robertlangenberg

### CODE REVIEW

File: Core/include/Acts/Seeding/BinnedSPGroup.hpp [hunk 1]
```diff
@@ -188,34 +193,46 @@ class BinnedSPGroupIterator {
...
-    outputIndex = 0;
-    bottomBinIndices = m_bottomBinFinder->findBins(phiIndex, zIndex, grid);
-    topBinIndices = m_topBinFinder->findBins(phiIndex, zIndex, grid);
+    customZorder = bins;
+    size_t this_zIndex = bins.empty() ? zIndex : bins.front();
+    outputIndex = grid->globalBinFromLocalBins({phiIndex, this_zIndex});
+    currentBin = NeighborhoodVector{
+        grid->globalBinFromLocalBins({phiIndex, this_zIndex})};
+    bottomBinIndices = m_bottomBinFinder->findBins(phiIndex, this_zIndex, grid);
+    topBinIndices = m_topBinFinder->findBins(phiIndex, this_zIndex, grid);
```
Comment (robertlangenberg) [inline Core/include/Acts/Seeding/BinnedSPGroup.hpp:210]: please add a short comment explaining what happens

**Review by robertlangenberg — CHANGES_REQUESTED** [review]
just add a few comments to make the code easier to understand and I'm happy :)

File: Examples/Algorithms/TrackFinding/src/SeedingAlgorithm.cpp [hunk 2]
```diff
@@ -87,6 +87,16 @@ ActsExamples::SeedingAlgorithm::SeedingAlgorithm(
...
+  if (m_cfg.seedFinderConfig.zBinsCustomLooping.size() != 0) {
+    for (size_t i = 1; i != m_cfg.gridConfig.zBinEdges.size(); i++) {
+      if (std::find(m_cfg.seedFinderConfig.zBinsCustomLooping.begin(),
+                    m_cfg.seedFinderConfig.zBinsCustomLooping.end(),
+                    i) == m_cfg.seedFinderConfig.zBinsCustomLooping.end()) {
+        throw std::invalid_argument("Inconsistent config zBinsCustomLooping");
+      }
+    }
+  }
+
```
Comment (robertlangenberg) [inline Examples/Algorithms/TrackFinding/src/SeedingAlgorithm.cpp:99]: please add comment what consistent/inconsistent means

File: Core/include/Acts/Seeding/BinnedSPGroup.hpp (hunk 1 above)
Comment (LuisFelipeCoelho) [reply Core/include/Acts/Seeding/BinnedSPGroup.hpp:210]: ok! I will add some comments

File: Examples/Algorithms/TrackFinding/src/SeedingAlgorithm.cpp (hunk 2 above)
Comment (LuisFelipeCoelho) [reply Examples/Algorithms/TrackFinding/src/SeedingAlgorithm.cpp]: I added some comments, please let me know if they are enough

**Review by robertlangenberg — APPROVED** [review]

**Pull request merged**